client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -557,6 +562,10 @@ public void flush() throws SegmentSealedException {
                 }
             } catch (Exception e) {
                 failConnection(e);
+                if (e instanceof RetriesExhaustedException) {
+                    //throw an exception to the external world that the flush failed due to RetriesExhaustedException
+                    Exceptions.sneakyThrow(e);
```

Comment:
what about adding a "return" ?
it won't have effect but it may make the code cleaner

Reply:
The intent here was to ensure the user knows that there was a connectivity issue for this synchronous API. Adding a return with a log message will imply the user will not know that some events were not written due to connectivity issues.

@tkaitchuck is there a better way to communicate this to the user of Pravega?

Reply:
Exceptions.sneakyThrow(e); throws an exception, and the execution of the method is aborted, so basically it is like we had a "return" after   Exceptions.sneakyThrow(e);

my concern is only about telling explicitly to the reader of the code that the execution is ending up there (also automatic flow execution tools like spotbugs may be happier)

not a big deal, we can keep it as it is now

Reply:
Ok, Thanks for clarifying it.
I have improved the code.